### PR TITLE
Reduce ExecutionState size and compute stacklimit correctly

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: src/runtime/ThreadLocal.cpp [hunk 1]
```diff
@@ -131,6 +142,45 @@ void ThreadLocal::initialize()
...
+#ifdef STACK_GROWS_DOWN
+    stackEndAddress = (char*)stackStartAddress - stackSize;
+#if defined(OS_DARWIN)
+    std::swap(stackStartAddress, stackEndAddress);
+#endif
+#else
+    stackEndAddress = (char*)stackStartAddress + stackSize;
+#endif
+#endif
+    stackSize = std::min(stackSize, (size_t)STACK_USAGE_LIMIT);
```
Comment (clover2123) [inline src/runtime/ThreadLocal.cpp:174]: It seems that `stackSize = std::min(stackSize, (size_t)STACK_USAGE_LIMIT);` is not used afterward.

File: src/runtime/ThreadLocal.cpp [hunk 2]
```diff
@@ -131,6 +142,45 @@ void ThreadLocal::initialize()
...
+
+#ifdef STACK_GROWS_DOWN
+    UNUSED_VARIABLE(stackEndAddress);
+    g_stackLimit = reinterpret_cast<size_t>(stackStartAddress) + STACK_FREESPACE_FROM_LIMIT;
```
Comment (clover2123) [inline src/runtime/ThreadLocal.cpp:178]: This limit means that we only use `STACK_FREESPACE_FROM_LIMIT` size for stack which is 256 KB,
Am I correct?

File: src/parser/ast/ReturnStatementNode.h [hunk 3]
```diff
@@ -65,10 +65,12 @@ class ReturnStatementNode : public StatementNode {
                 m_argument->generateExpressionByteCode(codeBlock, context, index);
 #endif
                 codeBlock->pushCode(ReturnFunctionSlowCase(ByteCodeLOC(m_loc.index), index), context, this->m_loc.index);
+                context->m_needsExtendedExecutionState = true;
             } else {
                 index = context->getRegister();
                 codeBlock->pushCode(LoadLiteral(ByteCodeLOC(m_loc.index), index, Value()), context, this->m_loc.index);
                 codeBlock->pushCode(ReturnFunctionSlowCase(ByteCodeLOC(m_loc.index), index), context, this->m_loc.index);
+                context->m_needsExtendedExecutionState = true;
```
Comment (clover2123) [inline src/parser/ast/ReturnStatementNode.h:73]: Is this marking in return statement also necessary?

File: src/runtime/ThreadLocal.cpp (hunk 2 above)
Comment (ksh8281) [reply src/runtime/ThreadLocal.cpp:178]: If real stack size is 8MB, we will use (8MB-STACK_FREESPACE_FROM_LIMIT).
remained space used by computing loc when stackoverflow is occured

File: src/parser/ast/ReturnStatementNode.h (hunk 3 above)
Comment (ksh8281) [reply src/parser/ast/ReturnStatementNode.h:73]: ReturnFunctionSlowCase opcode may needs ControlFlowRecord

File: src/runtime/ThreadLocal.cpp (hunk 1 above)
Comment (ksh8281) [reply src/runtime/ThreadLocal.cpp:174]: Oops my mistake

File: src/runtime/ThreadLocal.cpp (hunk 2 above)
Comment (clover2123) [reply src/runtime/ThreadLocal.cpp:179]: I see!

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**